data/src/main/java/com/linkedin/data/schema/PdlBuilder.java

```diff
@@ -149,7 +149,7 @@ PdlBuilder writeProperties(List<String> prefix, Map<String, Object> properties)
       {
         // Favor @x.y.z = "value" property encoding style over @x = { "y": { "z": "value" } }
         DataMap dm = (DataMap) value;
-        if (!dm.isEmpty())
+        if (!dm.isEmpty() && dm.size() == 1)
```

Comment:
Redundant. Just `dm.size() == 1` suffices.

Reply:
done. Initially thought dm could be null also which is not the case